README.md

```diff
@@ -189,28 +190,156 @@ When Silead touchscreen support was added to ACPI (PC) platforms, the
+Repeat this about 5 times to make sure you really have found
+the lowest / highest value the axis reports.
+
+Write down the lowest ABS_X value observed as x-min and the highest
+observed value (on the other edge) as x-max.
+
+Repeat this for ABS_Y and write down y-min and y-max.
+
+#### 5. Testing the parameters
+
+When running kernel 6.10 or later you can now test the found min/max
+values on the kernel commandline.
+
+This requires the ACPI Hardware ID (HID) for the touchscreen. Run
+`ls -d /sys/bus/i2c/devices/i2c-MSSL*`, this should output a single line
+ending in e.g. `i2c-MSSL1680:00`. The ACPI HID of the touchscreen is the bit
+after the `i2c-` and before the `:00` so `MSSL1680` in this case.
+
+This HID is the first argument for the `i2c_touchscreen_props` kernel
+commandline argument. To pass the found min/max values the following syntax
+is used:
+
+```
+i2c_touchscreen_props=<ACPI_HW_ID>:touchscreen-min-x=<x-min>:touchscreen-min-y=<y-min>:touchscreen-size-x=<x-max + 1>:touchscreen-size-y=<y-max + 1>
+```
+
+So if for example in step 4. you have found values of x-min=8, y-min=8,
+x-max=1919, y-max=1079 then the kernel commandline argument to pass would be:
+
+```
+i2c_touchscreen_props=MSSL1680:touchscreen-min-x=8:touchscreen-min-y=8:touchscreen-size-x=1920:touchscreen-size-y=1080
+```
+
+Edit your kernel configuration, e.g. `/boot/loader/entries/<current-kernel>.conf`,
+add this to your kernel commandline then reboot to test. After reboot run
+`sudo cat /proc/cmdline` to make sure that `i2c_touchscreen_props=...` is there
+and is correct.
+
+With some luck your touchscreen will now work right away. Somtimes the X/Y axis
+need be to swapped or inverted. Note swapping will automatically swap the size
+and min parameters, so those must not be swapped manually. So lets say you need
+to swap and invert the Y axis, you would then extend `i2c_touchscreen_props=`
+by adding `:touchscreen-swapped-x-y:touchscreen-inverted-y` after the size-x
+parameter, so you would get (abbreviated):
+
+```
+i2c_touchscreen_props=MSSL1680:touchscreen-min-x=...:touchscreen-size-y=1080:touchscreen-swapped-x-y:touchscreen-inverted-y
+```
+
+If one of the bezels around the display has a Windows logo chances are this
+can operate as a capacitive home-button. To test this add `:silead,home-button`
+after the existing `i2c_touchscreen_props=` parameters. After rebooting with
+`:silead,home-button` added to the parameters run `sudo evtest /dev/input/event#`
+and press the Windows logo if this reports `KEY_LEFTMETA` events then it works.
+
+If this does not work drop `silead,home-button` from your parameters since this
+may cause spurious `KEY_LEFTMETA` events when it is not supported.
+
+The linux touchscreen framework also supports some other properties that
+may be useful, see [touchscreen.yaml](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/input/touchscreen/touchscreen.yaml)
+for a full list.
```

Comment:
Do we need to mention how to determine the number of multi-touch points?
I believe we've seen devices with 5 and some with 10.

Reply:
Basically all but the very oldest Silead touchscreens only found on really cheap Allwinner A13 / A23 tablets can do 10 points. So I plan to submit a patch upstream to make 10 the default and drop all the boilerplate PROPERTY_ENTRY_U32("silead,max-fingers", 10) code from drivers/platform/x86/touchscreen_dmi.c I will submit a patch series for this upstream soon.